### ISSUE TITLE
non-english repos

### ISSUE THREAD

**Issue by boyeborg** [thechangelog/nightly#31]
Lately, there has been a lot of non-english repos (like [this 17-times champion](https://github.com/lib-pku/libpku)), mostly chinese from what I can tell. Since I don't speek chinese, these repos are only noise, and makes the newsletter less valuable to me. I usually reed changelog nightly on my phone (on my way to work), hence using google translate is not a viable option.

I don't want to be an advocate to remove repos with non-english readmes, since I believe a lot of cool stuff is happening outside of the english linguistical circles. However, I can immediately think of two ways of making the newsletter more readable to me:

1. Allow users to subscribe to a english-only newsletter or an interligual version.
2. Create a seperate section within the cahngelog nightly for non-english repos.

I think it's important to think thoroughly before implementing a solution. In the GitHub environment, non-english repositories can be seen as a minority, and just filtering them out might be a bad decision. Also, this is just a suggestion that would make the newsletter better for ME, and might not apply to everyone else. My suggestions are only ment as a conversation starter, as I think this is a conversation that's needed.

**Comment by jerodsanto** [comment]
@boyeborg this is something I've considered for a long time. I agree with you that the frequency of non-english repos are on the rise lately.

We already have rudimentary language detection support [in the code](https://github.com/thechangelog/nightly/blob/master/lib/repo.rb#L43-L45), so that part is relatively easy on the technical side. (granted, it's only used to _not_ tweet certain repos, which has a low risk failure state so I'm not sure how accurate it is. I'm sure there are jargon-filled enlish-language repos that it also determine are not english).

Both of your suggestions have merit, but I'd lean toward #1 for the best user experience. Another thought I had was to include a "translate" link next to descriptions that are determined to be non-english and link out to google translate with the data pre-populated. What do you think of that idea?

**Comment by jerodsanto** [comment]
@boyeborg polling the tweeters to see what they think about this. Feel free to vote there as well:

https://twitter.com/changelog/status/1125758373859876865

**Comment by boyeborg** [comment]
@jerodsanto Good thinking with the poll! I also think a translate link would be a great addition. 

I have given it some more thought, and I'm not that sure an english-only version is such a great idea after all. I believe that code and programming isn't (and shouldn't) be connected to one single spoken language or country, but rather a global an international way of exchanging thought, ideas and solutions (and even problems).
To create a product (changelog nightly english-only edition™) that removes a lot of important attributions to GitHub, and the programming community, just because they aren't in our desired language, is bad. We might say, or think, that by allowing the users to choose to subscribe to a interligual version, we include all repos of all languages, but I don't think this necessarily holds true. A lot of people would just subscribe to the english-only version, and never be exposed of non-english contribution.
Hence, I belive we should strive towards helping people understand the non-english repos rather than just filtering them out. Maybe there might be some repos that aren't readable to everyone, but I believe the occasional "noise" is a fair price to pay for the diversity gained by including repos of all languages.

**Comment by boyeborg** [comment]
An example of this is the attention 996.ICU got. It would be very bad to have a changelog nightly version with that filtered out. Although the main repo is in english, I believe a big part of changelog nightly is about picking up trends and upcoming repos (i.e. before they become popular and translated to english), hence including non-english repos makes a lot of sense.

**Comment by jerodsanto** [comment]
@boyeborg I agree with you and will look in to what it'll take to add translate links for non-english repos ✊

**Comment by jaraddowning** [comment]
I just wanted to add my $0.02. I came here to request the same as @boyeborg and have repos that are in languages that don't use latin characters filtered out. I see valid points made already and think having a translation available would be the perfect solution (in theory) as the value of the repos I cannot read would not be completely lost. Thanks @jerodsanto for doing what you do. I've got some (read very little) experience in translating and I'll see if I can help.

**Comment by jerodsanto** [comment]
Ok keep your 👀 peeled tonight for translate links next to non-english repos 🤞

**Comment by boyeborg** [comment]
@jerodsanto: Seems like there are a lot of english repos with translation links. Not that this is a problem for me, since I just don't click the translate-link if it's in english. Perhaps all repost should have translation links? That would make it easy for people who arn't fluent in english to translate english repos as well. It's quite easy to just change language once you are on the google translate website.

Another improvment could be to include the whole readme in the translation, if this is possible? I tried to URL-encode the readme of [wepe/efficient-decision-tree-notes](https://github.com/wepe/efficient-decision-tree-notes) and past the whole thing in the google translate URL. It seems to work fine, you can [see the result here](https://translate.google.com/#view=home&op=translate&sl=auto&tl=en&text=%23%20efficient-decision-tree-notes%0A%0A%E8%BF%99%E4%B8%AA%E7%AC%94%E8%AE%B0%E8%AE%B0%E5%BD%95%E9%AB%98%E6%95%88%E7%9A%84%E5%86%B3%E7%AD%96%E6%A0%91%E7%B3%BB%E5%88%97%E7%AE%97%E6%B3%95%EF%BC%8C%E4%B8%BB%E8%A6%81%E9%98%85%E8%AF%BB%E8%AE%BA%E6%96%87%EF%BC%8C%E7%BB%93%E5%90%88%E4%B8%80%E4%BA%9B%E5%BC%80%E6%BA%90%E6%A1%86%E6%9E%B6%EF%BC%8C%E5%B8%8C%E6%9C%9B%E5%9C%A8%E5%BC%84%E6%B8%85%E7%AE%97%E6%B3%95%E7%9A%84%E5%9F%BA%E7%A1%80%E4%B8%8A%EF%BC%8C%E5%B0%9D%E8%AF%95%E7%9D%80%E6%94%B9%E8%BF%9B%E7%AE%97%E6%B3%95%EF%BC%8C%E5%B0%9D%E8%AF%95%E7%9D%80%E5%B7%A5%E7%A8%8B%E5%AE%9E%E7%8E%B0%EF%BC%8E%0A%0A%E6%88%91%E4%BB%AC%E7%9F%A5%E9%81%93%EF%BC%8C%E7%9B%AE%E5%89%8D%E6%AF%94%E8%BE%83%E6%B5%81%E8%A1%8C%E7%9A%84%E4%B8%A4%E4%B8%AAGBDT%E5%BC%80%E6%BA%90%E6%A1%86%E6%9E%B6%E6%98%AFXGBoost%E5%92%8CLightGBM%EF%BC%8C%E6%97%A0%E8%AE%BA%E5%86%85%E5%AD%98%E5%8D%A0%E7%94%A8%E8%BF%98%E6%98%AF%E8%AE%A1%E7%AE%97%E9%80%9F%E5%BA%A6%EF%BC%8C%E5%AE%83%E4%BB%AC%E9%83%BD%E5%81%9A%E5%88%B0%E4%BA%86%E6%B7%8B%E6%BC%93%E5%B0%BD%E8%87%B4%EF%BC%8E%E5%9C%A8LightGBM%E7%9A%84%5BFeature%5D%28http%3A%2F%2Flightgbm.readthedocs.io%2Fen%2Flatest%2FFeatures.html%29%E4%B8%8A%E6%8F%90%E5%88%B0%E4%BA%86%EF%BC%8CXGBoost%E7%9A%84decision%20tree%E7%94%A8%E7%9A%84%E6%98%AFpre-sorted%20based%E7%9A%84%E7%AE%97%E6%B3%95%EF%BC%8C%E4%B9%9F%E5%B0%B1%E6%98%AF%E5%9C%A8tree%20building%E4%B9%8B%E5%89%8D%E5%AF%B9%E5%90%84%E7%BB%B4%E7%89%B9%E5%BE%81%E5%85%88%E6%8E%92%E5%BA%8F%EF%BC%8C%E4%BB%A3%E8%A1%A8%E6%80%A7%E7%9A%84%E7%AE%97%E6%B3%95%E6%98%AFSLIQ%5B1%5D%E5%92%8CSPRINT%5B2%5D%EF%BC%8E%E8%80%8CLightGBM%E7%9A%84decision%20tree%E6%98%AFhistogram%20based%E7%9A%84%E7%AE%97%E6%B3%95%EF%BC%8C%E4%B9%9F%E5%B0%B1%E6%98%AF%E5%85%88%E5%B0%86%E7%89%B9%E5%BE%81%E7%A6%BB%E6%95%A3%E5%8C%96%EF%BC%8C%E4%BB%A3%E8%A1%A8%E6%80%A7%E7%9A%84%E7%AE%97%E6%B3%95%E6%98%AFCLOUDS%5B3%5D%2CMcrank%5B4%5D%E5%92%8CMachado%5B5%5D%EF%BC%8E%0A%0ASLIQ%E5%92%8CSPRINT%E7%AE%97%E6%B3%95%E7%9A%84%E7%89%B9%E7%82%B9%E5%86%B3%E5%AE%9A%E4%BA%86%E6%A0%91%E7%94%9F%E9%95%BF%E7%9A%84%E6%96%B9%E5%BC%8F%E6%98%AFlevel-wise%28breadth-first%29%E7%9A%84%EF%BC%8C%E4%B8%8E%E4%B9%8B%E5%AF%B9%E5%BA%94%E7%9A%84%E6%98%AFleaf-wise%28depth-wise%EF%BC%8Cbest-wise%5B6%5D%29%E7%9A%84%E6%96%B9%E5%BC%8F%EF%BC%8CLightGBM%E6%AD%A3%E6%98%AF%E9%87%87%E7%94%A8leaf-wise%E7%9A%84%E6%96%B9%E5%BC%8F%EF%BC%8E%0A%0A%E5%86%85%E5%AE%B9%E5%A4%A7%E8%87%B4%E6%8C%89%E4%BB%A5%E4%B8%8B%E5%87%A0%E9%83%A8%E5%88%86%E5%B1%95%E5%BC%80%3A%0A%0A-%20%E5%9F%BA%E4%BA%8E%E7%89%B9%E5%BE%81%E9%A2%84%E6%8E%92%E5%BA%8F%E7%9A%84%E7%AE%97%E6%B3%95%5BSLIQ%5D%28https%3A%2F%2Fgithub.com%2Fwepe%2Fefficient-decision-tree-notes%2Fblob%2Fmaster%2FSLIQ.md%29%0A%0A-%20%E5%9F%BA%E4%BA%8E%E7%89%B9%E5%BE%81%E9%A2%84%E6%8E%92%E5%BA%8F%E7%9A%84%E7%AE%97%E6%B3%95%5BSPRINT%5D%28https%3A%2F%2Fgithub.com%2Fwepe%2Fefficient-decision-tree-notes%2Fblob%2Fmaster%2FSPRINT.md%29%0A%0A-%20%E5%9F%BA%E4%BA%8E%E7%89%B9%E5%BE%81%E7%A6%BB%E6%95%A3%E5%8C%96%E7%9A%84%E7%AE%97%E6%B3%95%5BCLOUDS%5D%28https%3A%2F%2Fgithub.com%2Fwepe%2Fefficient-decision-tree-notes%2Fblob%2Fmaster%2FClOUDS.md%29%0A%0A-%20%E7%A0%94%E7%A9%B6%E5%BC%80%E6%BA%90%E6%A1%86%E6%9E%B6%3A%20%5BLightGBM%EF%BC%8CXGBoost%5D%28https%3A%2F%2Fgithub.com%2Fwepe%2Fefficient-decision-tree-notes%2Fblob%2Fmaster%2FLightGBM%2520vs%2520XGBoost.md%29%0A%0A-%20%E8%87%AA%E5%B7%B1%E5%8A%A8%E6%89%8B%E5%AE%9E%E7%8E%B0%E4%B8%80%E4%B8%AA%E9%AB%98%E6%95%88%E5%86%B3%E7%AD%96%E6%A0%91%20%5Btgboost%5D%28https%3A%2F%2Fgithub.com%2Fwepe%2Ftgboost%29%0A%0A%0A%23%23%23%20%E5%8F%82%E8%80%83%E6%96%87%E7%8C%AE%0A%0A%0A%5B1%5D%20Mehta%2C%20Manish%2C%20Rakesh%20Agrawal%2C%20and%20Jorma%20Rissanen.%20%E2%80%9CSLIQ%3A%20A%20fast%20scalable%20classifier%20for%20data%20mining.%E2%80%9D%20International%20Conference%20on%20Extending%20Database%20Technology.%20Springer%20Berlin%20Heidelberg%2C%201996.%0A%0A%5B2%5D%20Shafer%2C%20John%2C%20Rakesh%20Agrawal%2C%20and%20Manish%20Mehta.%20%E2%80%9CSPRINT%3A%20A%20scalable%20parallel%20classifier%20for%20data%20mining.%E2%80%9D%20Proc.%201996%20Int.%20Conf.%20Very%20Large%20Data%20Bases.%201996.%0A%0A%5B3%5D%20Ranka%2C%20Sanjay%2C%20and%20V.%20Singh.%20%E2%80%9CCLOUDS%3A%20A%20decision%20tree%20classifier%20for%20large%20datasets.%E2%80%9D%20Proceedings%20of%20the%204th%20Knowledge%20Discovery%20and%20Data%20Mining%20Conference.%201998.%0A%0A%5B4%5D%20Machado%2C%20F.%20P.%20%E2%80%9CCommunication%20and%20memory%20efficient%20parallel%20decision%20tree%20construction.%E2%80%9D%20%282003%29.%0A%0A%5B5%5D%20Li%2C%20Ping%2C%20Qiang%20Wu%2C%20and%20Christopher%20J.%20Burges.%20%E2%80%9CMcrank%3A%20Learning%20to%20rank%20using%20multiple%20classification%20and%20gradient%20boosting.%E2%80%9D%20Advances%20in%20neural%20information%20processing%20systems.%202007.%0A%0A%5B6%5D%20Shi%2C%20Haijian.%20%E2%80%9CBest-first%20decision%20tree%20learning.%E2%80%9D%20Diss.%20The%20University%20of%20Waikato%2C%202007.%0A).

**Comment by jerodsanto** [comment]
Yeah, I noticed that the `whatlanguage` gem isn't really doing a great job at detecting english vs non. Passing the entire README to the translator is a cool idea, but my concern is that certain email clients (namely Gmail) will truncate email that is too "long", which I believe they measure by total page weight. Stuffing 20ish readmes into anchor tags might push us over that edge, which isn't something I'd like to deal with.

Having the translation link on every repo isn't a bad compromise, but maybe we need to move it to an icon or something less obtrusive in that case...

**Comment by boyeborg** [comment]
I see the problem with having a too "heavy" emails. The fact that the URLs are shortened to something like `https://email.changelog.com/t/...` doesn't help?

I agree with having a symbol og icon. Perhaps something like [this font awesome icon](https://fontawesome.com/icons/language?style=solid) would suffice? It would be nice if it somehow also indicated that it was a URL. Perhaps [the google translate logo](https://en.wikipedia.org/wiki/Google_Translate#/media/File:Google_Translate_logo.svg) would be a better option?

**Comment by jerodsanto** [comment]
That's a good point on Campaign Monitor rewriting those URLs for us. That definitely helps, but it assumes that we will have rewrites for those in perpetuity, which might not be 100% the case.

I like the idea of using the Google Translate logo, but curious what @codyjames thinks on the matter.

**Comment by codyjames** [comment]
Maybe a one color versions of the Google Translate logo if we can find a good one. 

Also, here is what Noun Project has: https://thenounproject.com/search/?q=translate

**Comment by jerodsanto** [comment]
I like this one quite a bit:

https://thenounproject.com/search/?q=translate&i=1436334#

However, it is a bit hard to read when it's just a tiny icon like this

<img width="612" alt="2019-05-30 at 12 12 PM" src="https://user-images.githubusercontent.com/8212/58650483-68890a80-82d4-11e9-8a05-128175b0c368.png">

**Comment by codyjames** [comment]
Yeah, I was worried about that with any of the staggered ones, including the Google Translate icon.

**Comment by codyjames** [comment]
Maybe something simple like this: https://thenounproject.com/search/?q=translate&i=41100

**Comment by jerodsanto** [comment]
How does this look?

<img width="430" alt="2019-05-30 at 2 28 PM" src="https://user-images.githubusercontent.com/8212/58658899-3b922300-82e7-11e9-80e2-447cf0f3ede8.png">

**Comment by codyjames** [comment]
The "A" looks a bit messed up and doesn't quite read as an "A". I think if we clean that up this would be fine.

**Comment by jerodsanto** [comment]
Okay I cleaned up both glyphs a bit so they're more clear:

<img width="409" alt="2019-05-31 at 9 59 AM" src="https://user-images.githubusercontent.com/8212/58714675-d6900900-838a-11e9-9fd8-905869614885.png">

The title and alt attribute are set to 'Translate' so if you're not sure and hover your mouse long enough you'll get the picture.

Also kinda neat (I think): I previously had the `fl` (from language) param set to `auto` so that Google Translate will guess what the source is. In addition to that, I changed the `tl` (to language) param to also be `auto` (previously was `en`) so that it will do its best guess of what language you want it translated to. It's hard to test how well that works since for me it always goes to English anyhow...

This _might_ help non-english speakers quickly translate the descriptions to their own locale as well. 🤞
